plugins/outputs/dynatrace/dynatrace_test.go

```diff
@@ -515,3 +515,52 @@ func TestSendCounterMetricWithoutTags(t *testing.T) {
 	err = d.Write(metrics)
 	require.NoError(t, err)
 }
+
+var warnfCalledTimes int
+
+type loggerStub struct {
+	testutil.Logger
+}
+
+func (l loggerStub) Warnf(format string, args ...interface{}) {
+	warnfCalledTimes += 1
+}
+
+func TestSendUnsupportedMetric(t *testing.T) {
+	warnfCalledTimes = 0
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Fatal("should not export because the only metric is an invalid type")
+	}))
+	defer ts.Close()
+
+	d := &Dynatrace{}
+
+	logStub := loggerStub{}
+
+	d.URL = ts.URL
+	d.APIToken = "123"
+	d.Log = logStub
+	err := d.Init()
+	require.NoError(t, err)
+	err = d.Connect()
+	require.NoError(t, err)
+
+	// Init metrics
+
+	m1 := metric.New(
+		"mymeasurement",
+		map[string]string{},
+		map[string]interface{}{"unsupported": "type"},
+		time.Date(2010, time.November, 10, 23, 0, 0, 0, time.UTC),
+	)
+
+	metrics := []telegraf.Metric{m1}
+
+	err = d.Write(metrics)
+	require.NoError(t, err)
+	require.Equal(t, 1, warnfCalledTimes)
+
+	err = d.Write(metrics)
+	require.NoError(t, err)
+	require.Equal(t, 1, warnfCalledTimes)
```

Comment:
```suggestion
	require.Equal(t, 1, warnfCalledTimes) // each metric name is logged only once
```
To make it more obvious.